### DISCUSSION TITLE
Simple hack to have stable and offline Requestly front-end

### DISCUSSION THREAD

**Discussion by manasan3010** [requestly/requestly#594] [General]
Requestly is a good extension but you can't edit rules offline and the webapp also introduce betaking changes often so here is a simple solution to men who prefers stability over style.

1. Extract the below archive and point your local webserver to the decompressed directory
2. Edit your `hosts` file to resolve `app.requestly.io` -> `127.0.0.1`
3. Instruct your webserver to serve index.html for 404 status code
4. Clear HSTS cache for `app.requestly.io` domain
5. Use `https` to access `app.requestly.io` and everything should work now offline!

[app.requestly.io.zip](https://github.com/requestly/requestly/files/11300623/app.requestly.io.zip)

**Comment by ashishsangle707** [comment]
@manasan3010,
This is Ashish here from Requestly, Product team. 

Thank you for sharing your solution for using Requestly offline. We are curious to know why you prefer using Requestly offline?

**Reply by Zorzi23** [reply]
Hello! I prefer using Requestly offline because my development environment lacks internet access. Having an offline version allows me to seamlessly integrate it into my workflow, ensuring uninterrupted productivity even when I'm not connected to the internet.

**Reply by filefi** [reply]
I'm also curious to know why a tool called "Local-first Developer Tool" cannot be used in a local area network.

**Reply by Luthien-in-edhil** [reply]
I came across Requestly trying to find an alternative for ModHeader-turned-malware. 
Installed it. 
Click "Modify Headers" --> Immediately opens an external website. 
Removed it.

@ashishsangle707 - I'm aware that this behaviour doesn't necessarily mean that the addon can't be trusted. But it doesn't inspire a lot of trust either: like @filefi says, why does an addon that can perfectly well do its job locally take me to an external site?  

Why not keep the local functionality strictly within the browser and add the app.request.io functionality as an optional  extra?

**Reply by sagarsoni7** [reply]
> I came across Requestly trying to find an alternative for ModHeader-turned-malware. Installed it. Click "Modify Headers" --> Immediately opens an external website. Removed it.
> 
> @ashishsangle707 - I'm aware that this behaviour doesn't necessarily mean that the addon can't be trusted. But it doesn't inspire a lot of trust either: like @filefi says, why does an addon that can perfectly well do its job locally take me to an external site?
> 
> Why not keep the local functionality strictly within the browser and add the app.request.io functionality as an optional extra?

Are you running `sh install.sh local` and `sh build.sh local`? It should generate the local builds and not take to external site.
